Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift

```diff
@@ -30,217 +25,82 @@ internal func docConfigFilter(forNamespace namespace: MongoNamespace,
-            self.isPaused = try values.decode(Bool.self, forKey: .isPaused)
-        }
-
-        // to BSON document
-        func encode(to encoder: Encoder) throws {
-            var container = encoder.container(keyedBy: CodingKeys.self)
-
-            try container.encode(documentId, forKey: .documentId)
-
-            // verify schema version
-            try container.encode(1 as Int32, forKey: .schemaVersion)
-
-            try container.encode(namespace, forKey: .namespace)
-            try container.encode(lastResolution, forKey: .lastResolution)
-
-            if let lastKnownRemoteVersion = lastKnownRemoteVersion {
-                try container.encode(lastKnownRemoteVersion, forKey: .lastKnownRemoteVersion)
-            }
-
-            if let uncommittedChangeEvent = uncommittedChangeEvent {
-                let changeEventDoc = try BSONEncoder().encode(uncommittedChangeEvent)
-                // TODO: This may put the doc above the 16MiB but ignore for now.
-                try container.encode(
-                    Binary.init(data: changeEventDoc.rawBSON, subtype: Binary.Subtype.generic),
-                    forKey: .uncommittedChangeEvent
-                )
-            }
-
-            try container.encode(isStale, forKey: .isStale)
-            try container.encode(isPaused, forKey: .isPaused)
-        }
-
-        required init(namespace: MongoNamespace,
-                      documentId: HashableBSONValue,
-                      lastUncommittedChangeEvent: ChangeEvent<Document>?,
-                      lastResolution: Int64,
-                      lastKnownRemoteVersion: Document?,
-                      isStale: Bool,
-                      isPaused: Bool) {
-            self.namespace = namespace
-            self.documentId = documentId
-            self.uncommittedChangeEvent = lastUncommittedChangeEvent
-            self.lastResolution = lastResolution
-            self.lastKnownRemoteVersion = lastKnownRemoteVersion
-            self.isStale = isStale
-            self.isPaused = isPaused
-        }
-
-        static func == (lhs: CoreDocumentSynchronization.Config,
-                        rhs: CoreDocumentSynchronization.Config) -> Bool {
-            return lhs.documentId == rhs.documentId
-        }
-
-        func hash(into hasher: inout Hasher) {
-            documentId.hash(into: &hasher)
-        }
-
+final class CoreDocumentSynchronization: Codable, Hashable {
+    enum CodingKeys: String, CodingKey {
+        // These are snake_case because we are trying to keep the internal
+        // representation consistent across platforms
+        case namespace = "namespace", documentId = "document_id",
+        uncommittedChangeEvent = "last_uncommitted_change_event",
+        lastResolution = "last_resolution", lastKnownRemoteVersion = "last_known_remote_version",
+        isStale = "is_stale", isPaused = "is_paused", schemaVersion = "schema_version"
+        case docsColl = "docs_coll"
```

Comment:
[task] add a note to https://docs.google.com/document/d/1vaeI7m8Gv7zWDh0yETfqgX3JOeiX88k2Y9STCwsAKp0/edit?usp=drive_web&ouid=101429542541116054715 indicating that this is different from the on-device representation on Android. There is no `docs_coll` on Android.